.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,57 @@
+name: Benchmark
+
+on:
+  push:
+    branches:
+      - version-3
+
+  pull_request:
+    branches:
+      - version-3
+
+permissions:
+  id-token: write # required for OIDC
+  contents: read
+
+jobs:
+  benchmark:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby: [2.6, 2.7, 3.1, 3.2, jruby-9.2]
```

Comment:
We support 2.3+ and jruby 9.1+

Reply:
This was based on top usage (plus adding a few more recent rubies) - the goal was not to produce too many metrics overall.